packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -282,17 +319,34 @@ class _BottomSheetState extends State<BottomSheet> {
     return false;
   }
 
+  void _handleDragHandleHover(bool hovering) {
+    if (hovering != _isHoveringDragHandle) {
+      setState(() {
+        _isHoveringDragHandle = hovering;
+      });
+    }
+  }
+
+  Set<MaterialState> get dragHandleMaterialState {
```

Comment:
It might be slightly simpler to:
- Replace _isDragging and _isHovering with dragHandleState.contains(MaterialState.hovered), dragHandleState.add(MaterialState.hovered) etc. You could tidy all that up with getters and setters that mutated the dragHandleState set.
- Use MaterialState.dragged for the drag state. If necessary, always set dragged and focused in tandem.